ui/src/cloud/apis/demodata.ts

```diff
@@ -0,0 +1,93 @@
+// Libraries
+import {get} from 'lodash'
+import * as api from 'src/client'
+import AJAX from 'src/utils/ajax'
+
+//Utils
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {isDemoData} from 'src/cloud/utils/filterDemoData'
+
+//Types
+import {Bucket} from 'src/types'
+import {LIMIT} from 'src/resources/constants'
+
+const baseURL = '/api/v2/experimental/sampledata'
+
+export const getDemoDataBuckets = async (): Promise<Bucket[]> => {
+  try {
+    const {data} = await AJAX({
+      method: 'GET',
+      url: `${baseURL}/buckets`,
+    })
+
+    // if sampledata endpoints are not available in a cluster
+    // gateway responds with a list of links where 'buckets' field is a string
+    const buckets = get(data, 'buckets', false)
+    if (!buckets || !Array.isArray(buckets)) {
+      throw new Error('Could not reach demodata endpoint')
+    }
+
+    return buckets.filter(b => b.type == 'user') as Bucket[] // remove returned _tasks and _monitoring buckets
+  } catch (error) {
+    console.error(error)
+    throw error
+  }
+}
+
+export const getDemoDataBucketMembership = async (
+  bucketID: string,
+  userID: string
+) => {
+  try {
+    const response = await AJAX({
+      method: 'POST',
+      url: `${baseURL}/buckets/${bucketID}/members`,
+      data: {userID},
+    })
+
+    if (response.status === '200') {
+      // a failed or successful membership POST to sampledata should return 204
```

Comment:
is `204` the only good response? would it make sense to invert this?

```ts
if (response.status !== '204') {
  throw new Error('Could not reach demodata endpoint')
}

Reply:
I think this part is definitely going to change, and I think it's bad right now. Will ask for review of everything I've added as part of this PR when I make a PR to flip feature flag